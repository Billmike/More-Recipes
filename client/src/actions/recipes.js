import axios from 'axios';
import {
  ADD_RECIPE,
  GET_RECIPES,
  GET_ONE_RECIPE,
  GET_USER_RECIPES,
  ADD_FAVORITE_RECIPE,
  FETCH_FAVORITE_RECIPES,
  EDIT_RECIPE,
  REMOVE_RECIPE,
  ADD_REVIEW,
  UPVOTE_RECIPE,
  DOWNVOTE_RECIPE
} from './types';
import './toastrConfig';

export const addRecipe = recipe => ({
  type: ADD_RECIPE,
  recipe
});

export const getAllRecipes = (recipes) => {
  return {
    type: GET_RECIPES,
    recipes,
  };
};

export const getOneRecipe = (recipe) => {
  return {
    type: GET_ONE_RECIPE,
    id: recipe.id,
    recipe
  };
};

export const getUserRecipe = (userRecipe) => {
  return {
    type: GET_USER_RECIPES,
    userRecipe,
  };
};

export const addFavorites = (favoriteRecipes) => {
  return {
    type: ADD_FAVORITE_RECIPE,
    favoriteRecipes
  };
};

export const fetchFavorites = (favoriteRecipes) => {
  return {
    type: FETCH_FAVORITE_RECIPES,
    favoriteRecipes,
  };
};

export const editRecipe = (id, updates) => ({
  type: EDIT_RECIPE,
  id,
  updates,
});

export const removeRecipe = ({ id } = {}) => ({
  type: REMOVE_RECIPE,
  id
});

export const addReview = (review) => {
  return {
    type: ADD_REVIEW,
    review,
  };
};

export const upVoteRecipe = (id) => {
  return {
    type: UPVOTE_RECIPE,
    id,
  };
};

export const downVoteRecipe = (id) => {
  return {
    type: DOWNVOTE_RECIPE,
    id,
  };
};

export const startGetAllRecipes = () => {
  return (dispatch) => {
    axios.get('http://localhost:8000/api/v1/recipes')
      .then((res) => {
        dispatch(getAllRecipes(res.data.recipeData));
      })
      .catch((error) => {
        console.log(error);
      });
  };
}

export const startAddRecipe = (recipeData = {}) => {
  return (dispatch) => {
    const {
      name = '',
      description = '',
      img_url = 'no-img-here',
      category = '',
      ingredients = '',
      instructions = ''
    } = recipeData;

    const recipe = {
      name,
      description,
      img_url,
      category,
      ingredients,
      instructions,
    };
    axios.post('http://localhost:8000/api/v1/recipes', recipe)
      .then(() => {
        toastr.success('Recipe added successfully.');
        dispatch(addRecipe({
          id: recipe.id,
          ...recipe
        }));
      });
  };
};

export const startGetUserRecipes = () => {
  return (dispatch) => {
    axios.get('http://localhost:8000/api/v1/users/recipes')
      .then((res) => {
        dispatch(getUserRecipe(res.data.userRecipe));
      })
      .catch((error) => {
        console.log(error);
      });
  };
};

export const startEditRecipe = (id, updates) => {
  return (dispatch) => {
    axios.put(`http://localhost:8000/api/v1/recipes/${id}/modify`, updates)
      .then(() => {
        toastr.success('Recipe edited successfully.');
        dispatch(editRecipe(id, updates));
      })
      .catch((error) => {
        console.log('_<_<_<_<_<__<_', error);
      });
  };
};

export const startRemoveRecipe = (id) => {
  return (dispatch) => {
    axios.delete(`http://localhost:8000/api/v1/recipes/${id}`)
      .then(() => {
        toastr.success('Recipe deleted successfully.');
        dispatch(removeRecipe(id));
      })
      .catch((error) => {
        console.log(error);
      });
  };
};

export const startGetOneRecipe = (id) => {
  return (dispatch) => {
    axios.get(`http://localhost:8000/api/v1/recipes/${id}`)
      .then((res) => {
        dispatch(getOneRecipe(res.data.recipeData));
      })
      .catch((error) => {
        console.log(error);
      });
  };
};

export const startAddFavoriteRecipes = (id) => {
  return (dispatch) => {
    axios.post(`http://localhost:8000/api/v1/recipes/${id}/favorites`)
      .then((res) => {
        toastr.success(res.data.message);
        return dispatch(addFavorites(res.data));
      })
      .catch((err) => {
        if (err.response.data
          .message === 'You need to be logged in to perform this action.') {
          toastr.error(err.response.data.message);
        } else if (err.response.data
          .message === 'Sorry, you cannot perform this action on your own recipe.') {
          toastr.error(err.response.data.message);
        }
      });
  };
};

export const startGetUserFavorites = (id) => {
  return (dispatch) => {
    axios.get(`http://localhost:8000/api/v1/users/${id}/favorites`)
      .then((res) => {
        dispatch(fetchFavorites(res.data));
      })
      .catch((err) => {
        console.log(err);
      });
  };
};

export const startUpvoteRecipe = (id) => {
  return (dispatch) => {
    axios.post(`http://localhost:8000/api/v1/recipes/${id}/votes/upvote`)
      .then((res) => {
        toastr.success(res.data.message);
      })
      .catch((err) => {
        if (err.response.data
          .message === 'You need to be logged in to perform this action.') {
          toastr.error(err.response.data.message);
        } else if (err.response.data
          .message === 'You already upvoted on this recipe') {
          toastr.warning(err.response.data.message);
        }
      });
  };
};

export const startDownVoteRecipe = (id) => {
  return (dispatch) => {
    axios.post(`http://localhost:8000/api/v1/recipes/${id}/votes/downvote`)
      .then((res) => {
        toastr.success(res.data.message);
      })
      .catch((err) => {
        if (err.response.data
          .message === 'You need to be logged in to perform this action.') {
          toastr.error(err.response.data.message);
        } else if (err.response.data
          .message === 'You already downvoted on this recipe') {
          toastr.warning(err.response.data.message);
        }
      });
  };
};

export const startAddReview = (id, data) => {
  return (dispatch) => {
    axios.post(`http://localhost:8000/api/v1/recipes/${id}/reviews`, data)
      .then((res) => {
        dispatch(addReview(res.data));
      })
      .catch((err) => {
        if (err.response.data
          .message === 'You need to be logged in to perform this action.') {
          toastr.error(err.response.data.message);
        } else if (err.response.data
          .message === 'You can\'t post an empty review. Please, enter a happy review for this recipe.') {
          toastr.warning(err.response.data.message);
        }
      });
  };
};
