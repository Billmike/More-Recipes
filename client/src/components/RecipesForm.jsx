import React, { Component } from 'react';
import { connect } from 'react-redux';
import validateAddRecipe from '../../../server/middleware/validateAddRecipe';

class RecipesForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      name: this.props.recipe ? this.props.recipe.name : '',
      description: this.props.recipe ? this.props.recipe.description : '',
      img_url: 'no-img-here',
      category: this.props.recipe ? this.props.recipe.category : '',
      ingredients: this.props.recipe ? this.props.recipe.ingredients : '',
      instructions: this.props.recipe ? this.props.recipe.instructions : '',
      error: ''
    };
  }

  onDescriptionChange = e => {
    const description = e.target.value;
    this.setState(() => ({ description }));
  };

  onNameChange = e => {
    const name = e.target.value;
    this.setState(() => ({ name }));
  };

  onImgSet = e => {
    const img_url = e.target.value;
    this.setState(() => ({ img_url }));
  };

  onCategorySet = e => {
    const category = e.target.value;
    this.setState(() => ({ category }));
  };

  onIngredientsSet = e => {
    const ingredients = e.target.value;
    this.setState(() => ({ ingredients }));
  };

  onInstructionsSet = e => {
    const instructions = e.target.value;
    this.setState(() => ({ instructions }));
  };

  onSubmit = event => {
    event.preventDefault();
    this.props.onSubmit({
      name: this.state.name,
      description: this.state.description,
      img_url: this.state.img_url,
      category: this.state.category,
      ingredients: this.state.ingredients,
      instructions: this.state.instructions
    });
  };

  render() {
    return (
      <div>
        <main className="container">
          {this.state.error && <p>{this.state.error}</p>}
          <form className="needs-validation" onSubmit={this.onSubmit}>
            <div className="form-group">
              <label className="recipe-name" htmlFor="recipename">
                Recipe Name
              </label>
              <input
                className="form-control recipe-name"
                id="recipename"
                aria-describedby="recipeNameHelp"
                value={this.state.name}
                onChange={this.onNameChange}
                required
              />
              <small id="recipeNameHelp" className="form-text">
                Enter a name for your recipe
              </small>
            </div>
            <div className="form-row">
              <div className="form-group col-8">
                <label className="recipe-desc" htmlFor="inputDescription">
                  Description
                </label>
                <input
                  className="form-control"
                  id="descriptionTextArea"
                  rows="3"
                  value={this.state.description}
                  onChange={this.onDescriptionChange}
                  required
                />
                <small id="descriptionHelp" className="form-text">
                  Enter a short description for your recipe.
                </small>
              </div>
              <div className="form-group col-4 upload-btn-wrapper">
                <button className="btns">Upload an Image</button>
                <input type="file" name="myfile" />
              </div>
            </div>
            <div className="form-group">
              <label className="recipe-cat" htmlFor="recipename">
                Category
              </label>
              <input
                className="form-control"
                id="recipename"
                aria-describedby="recipeNameHelp"
                value={this.state.category}
                onChange={this.onCategorySet}
                required
              />
              <small id="categoryHelp" className="form-text">
                Category could be Breakfast, Lunch, Dessert or any other fun one
                of your choice.
              </small>
            </div>
            <div className="form-row">
              <div className="form-group col-md-6">
                <label className="recipe-ingredients" htmlFor="recipename">
                  Ingredients
                </label>
                <textarea
                  className="form-control large-text"
                  id="recipename"
                  aria-describedby="recipeNameHelp"
                  value={this.state.ingredients}
                  onChange={this.onIngredientsSet}
                  required
                />
                <small id="recipeNameHelp" className="form-text">
                  Enter your recipe ingredients, separated by Commas.
                </small>
              </div>
              <div className="form-group col-md-6">
                <label
                  className="recipe-instructions"
                  htmlFor="inputDescription"
                >
                  Instructions
                </label>
                <textarea
                  className="form-control large-text"
                  id="recipeDesc"
                  rows="3"
                  value={this.state.instructions}
                  onChange={this.onInstructionsSet}
                  required
                />
                <small id="descriptionHelp" className="form-text">
                  Enter your step-by-step Instructions.
                </small>
              </div>
            </div>
            <div className="form-group">
              <button className="btn btn-primary my-btn button-font btn-lg btn-block">
                Submit Recipe
              </button>
            </div>
          </form>
        </main>
      </div>
    );
  }
}

export default RecipesForm;
