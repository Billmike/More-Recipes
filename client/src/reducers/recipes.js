const recipeDefaultState = [];

export default (state = recipeDefaultState, action) => {
  switch (action.type) {
    case 'ADD_RECIPE':
      return [
        ...state,
        action.recipe,
      ];
    default:
      return state;
  }
};
