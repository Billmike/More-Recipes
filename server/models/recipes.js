const recipes = [
  {
    id: 1,
    name: 'Banana milk shake',
    description: 'Probably the best milk shake you ever had in your entire life',
    category: 'Dessert',
    ingredients: ['Milk', 'Banana', 'Olive oil'],
    instructions: ['Blend the banana properly', 'Filter and shake'],
    upVote: 0,
    downVote: 0,
    favorite: 0,
    reviews: [
      {
        review: '',
      },
    ],
  },
  {
    id: 2,
    name: 'Banana milk shake',
    description: 'Probably the best milk shake you ever had in your entire life',
    category: 'Dessert',
    ingredients: ['Milk', 'Banana', 'Olive oil'],
    instructions: ['Blend the banana properly', 'Filter and shake'],
    upVote: 0,
    downVote: 0,
    favorite: 0,
    reviews: [
      {
        review: '',
      },
    ],
  },
  {
    id: 3,
    name: 'Banana milk shake',
    description: 'Probably the best milk shake you ever had in your entire life',
    category: 'Dessert',
    ingredients: ['Milk', 'Banana', 'Olive oil'],
    instructions: ['Blend the banana properly', 'Filter and shake'],
    upVote: 0,
    downVote: 0,
    favorite: 0,
    reviews: [
      {
        review: '',
      },
    ],
  },
];

export default recipes;
