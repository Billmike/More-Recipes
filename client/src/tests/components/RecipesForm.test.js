import React from 'react';
import { shallow } from 'enzyme';
import RecipesForm from '../../components/RecipesForm';
import recipes from '../fixtures/recipes';

test('Should render the RecipesForm to the screen', () => {
	const wrapper = shallow(<RecipesForm />);
	expect(wrapper).toMatchSnapshot();
});

test('Should render the form with recipes data', () => {
	const wrapper = shallow(<RecipesForm recipe={recipes[0]} />);
	expect(wrapper).toMatchSnapshot();
});

test('Should set Recipe name on Input change', () => {
	const value = 'Fried Chicken';
	const wrapper = shallow(<RecipesForm />);
	wrapper.find('input').at(0).simulate('change', {
		target: { value }
	});
	expect(wrapper.state('name')).toBe(value);
	expect(wrapper).toMatchSnapshot();
});

test('Should set recipe description on input change', () => {
	const value = 'Amazing Fried Chicken';
	const wrapper = shallow(<RecipesForm />);
	wrapper.find('textarea').at(0).simulate('change', {
		target: { value }
	});
	expect(wrapper.state('description')).toBe(value);
	expect(wrapper).toMatchSnapshot();
});

test('Should set recipe category on input change', () => {
	const value = 'Lunch';
	const wrapper = shallow(<RecipesForm />);
	wrapper.find('input').at(2).simulate('change', {
		target: { value }
	});
	expect(wrapper.state('category')).toBe(value);
	expect(wrapper).toMatchSnapshot();
});

test('Should set the recipe ingredients on input change', () => {
	const value = ['Curry', 'Thyme', 'Maggi', 'Salt'];
	const wrapper = shallow(<RecipesForm />);
	wrapper.find('input').at(3).simulate('change', {
		target: { value }
	});
	expect(wrapper.state('ingredients')).toBe(value);
	expect(wrapper).toMatchSnapshot();
});

test('Should set the instructions on input change', () => {
	const value = ['Parboil the rice for 20mins', 'Boil the yam for 1hr.', 'Do not set the kitchen on fire.'];
	const wrapper = shallow(<RecipesForm />);
	wrapper.find('textarea').at(1).simulate('change', {
		target: { value }
	});
	expect(wrapper.state('instructions')).toBe(value);
	expect(wrapper).toMatchSnapshot();
});

test('Should call onSubmit prop for valid form submission', () => {
	const onSubmitSpy = jest.fn();
	const wrapper = shallow(<RecipesForm recipe={recipes[0]} onSubmit={onSubmitSpy} />);
	wrapper.find('form').simulate('submit', {
		preventDefault: () => { }
	});
	expect(onSubmitSpy).toHaveBeenLastCalledWith({
		name: recipes[0].name,
		description: recipes[0].description,
		category: recipes[0].category,
		img_url: recipes[0].img_url,
		ingredients: recipes[0].ingredients,
		instructions: recipes[0].instructions,
	});
});
