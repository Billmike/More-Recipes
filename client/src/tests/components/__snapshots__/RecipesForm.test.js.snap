// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the RecipesForm to the screen 1`] = `
<div>
  <main
    className="container"
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          htmlFor="recipename"
        >
          Recipe Title
        </label>
        <input
          aria-describedby="recipeNameHelp"
          className="form-control"
          id="recipename"
          onChange={[Function]}
          type="text"
          value=""
        />
        <small
          className="form-text text-muted"
          id="recipeNameHelp"
        >
          Enter a short, catchy title for your Recipe.
        </small>
      </div>
      <div
        className="form-row"
      >
        <div
          className="form-group col-8"
        >
          <label
            htmlFor="inputDescription"
          >
            Description
          </label>
          <textarea
            className="form-control"
            id="descriptionTextArea"
            onChange={[Function]}
            rows="3"
            value=""
          />
          <small
            className="form-text text-muted"
            id="descriptionHelp"
          >
            Try to keep your description short and interesting.
          </small>
        </div>
        <div
          className="form-group col-4 upload-btn-wrapper"
        >
          <button
            className="btns"
          >
            Upload an Image
          </button>
          <input
            name="myfile"
            type="file"
          />
        </div>
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="recipename"
        >
          Category
        </label>
        <input
          aria-describedby="recipeNameHelp"
          className="form-control"
          id="recipename"
          onChange={[Function]}
          placeholder="Breakfast, Dinner, Dessert, Lunch"
          type="text"
          value=""
        />
        <small
          className="form-text text-muted"
          id="recipeNameHelp"
        >
          Enter a short, catchy title for your Recipe.
        </small>
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="recipename"
        >
          Ingredients
        </label>
        <input
          aria-describedby="recipeNameHelp"
          className="form-control"
          id="recipename"
          onChange={[Function]}
          type="text"
          value={Array []}
        />
        <small
          className="form-text text-muted"
          id="recipeNameHelp"
        >
          Enter your recipe ingredients, separated by Commas.
        </small>
      </div>
      <div
        className="form-row"
      >
        <div
          className="form-group col-8"
        >
          <label
            htmlFor="inputDescription"
          >
            Instructions
          </label>
          <textarea
            className="form-control"
            id="descriptionTextArea"
            onChange={[Function]}
            rows="3"
            value={Array []}
          />
          <small
            className="form-text text-muted"
            id="descriptionHelp"
          >
            Enter your step-by-step Instructions.
          </small>
        </div>
        <div
          className="form-group col-4"
        >
          <button
            className="btn btn-outline-success btn-add"
          >
            Submit Recipe
          </button>
        </div>
      </div>
    </form>
  </main>
</div>
`;

exports[`Should render the form with recipes data 1`] = `
<div>
  <main
    className="container"
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          htmlFor="recipename"
        >
          Recipe Title
        </label>
        <input
          aria-describedby="recipeNameHelp"
          className="form-control"
          id="recipename"
          onChange={[Function]}
          type="text"
          value="Fried Rice"
        />
        <small
          className="form-text text-muted"
          id="recipeNameHelp"
        >
          Enter a short, catchy title for your Recipe.
        </small>
      </div>
      <div
        className="form-row"
      >
        <div
          className="form-group col-8"
        >
          <label
            htmlFor="inputDescription"
          >
            Description
          </label>
          <textarea
            className="form-control"
            id="descriptionTextArea"
            onChange={[Function]}
            rows="3"
            value="Awesome fried rice"
          />
          <small
            className="form-text text-muted"
            id="descriptionHelp"
          >
            Try to keep your description short and interesting.
          </small>
        </div>
        <div
          className="form-group col-4 upload-btn-wrapper"
        >
          <button
            className="btns"
          >
            Upload an Image
          </button>
          <input
            name="myfile"
            type="file"
          />
        </div>
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="recipename"
        >
          Category
        </label>
        <input
          aria-describedby="recipeNameHelp"
          className="form-control"
          id="recipename"
          onChange={[Function]}
          placeholder="Breakfast, Dinner, Dessert, Lunch"
          type="text"
          value="Lunch"
        />
        <small
          className="form-text text-muted"
          id="recipeNameHelp"
        >
          Enter a short, catchy title for your Recipe.
        </small>
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="recipename"
        >
          Ingredients
        </label>
        <input
          aria-describedby="recipeNameHelp"
          className="form-control"
          id="recipename"
          onChange={[Function]}
          type="text"
          value={
            Array [
              "Peppr",
              "Tomatoes",
              "Onions",
            ]
          }
        />
        <small
          className="form-text text-muted"
          id="recipeNameHelp"
        >
          Enter your recipe ingredients, separated by Commas.
        </small>
      </div>
      <div
        className="form-row"
      >
        <div
          className="form-group col-8"
        >
          <label
            htmlFor="inputDescription"
          >
            Instructions
          </label>
          <textarea
            className="form-control"
            id="descriptionTextArea"
            onChange={[Function]}
            rows="3"
            value={
              Array [
                "COok the food well",
                "Make no assumptions",
              ]
            }
          />
          <small
            className="form-text text-muted"
            id="descriptionHelp"
          >
            Enter your step-by-step Instructions.
          </small>
        </div>
        <div
          className="form-group col-4"
        >
          <button
            className="btn btn-outline-success btn-add"
          >
            Submit Recipe
          </button>
        </div>
      </div>
    </form>
  </main>
</div>
`;

exports[`Should set Recipe name on Input change 1`] = `
<div>
  <main
    className="container"
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          htmlFor="recipename"
        >
          Recipe Title
        </label>
        <input
          aria-describedby="recipeNameHelp"
          className="form-control"
          id="recipename"
          onChange={[Function]}
          type="text"
          value="Fried Chicken"
        />
        <small
          className="form-text text-muted"
          id="recipeNameHelp"
        >
          Enter a short, catchy title for your Recipe.
        </small>
      </div>
      <div
        className="form-row"
      >
        <div
          className="form-group col-8"
        >
          <label
            htmlFor="inputDescription"
          >
            Description
          </label>
          <textarea
            className="form-control"
            id="descriptionTextArea"
            onChange={[Function]}
            rows="3"
            value=""
          />
          <small
            className="form-text text-muted"
            id="descriptionHelp"
          >
            Try to keep your description short and interesting.
          </small>
        </div>
        <div
          className="form-group col-4 upload-btn-wrapper"
        >
          <button
            className="btns"
          >
            Upload an Image
          </button>
          <input
            name="myfile"
            type="file"
          />
        </div>
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="recipename"
        >
          Category
        </label>
        <input
          aria-describedby="recipeNameHelp"
          className="form-control"
          id="recipename"
          onChange={[Function]}
          placeholder="Breakfast, Dinner, Dessert, Lunch"
          type="text"
          value=""
        />
        <small
          className="form-text text-muted"
          id="recipeNameHelp"
        >
          Enter a short, catchy title for your Recipe.
        </small>
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="recipename"
        >
          Ingredients
        </label>
        <input
          aria-describedby="recipeNameHelp"
          className="form-control"
          id="recipename"
          onChange={[Function]}
          type="text"
          value={Array []}
        />
        <small
          className="form-text text-muted"
          id="recipeNameHelp"
        >
          Enter your recipe ingredients, separated by Commas.
        </small>
      </div>
      <div
        className="form-row"
      >
        <div
          className="form-group col-8"
        >
          <label
            htmlFor="inputDescription"
          >
            Instructions
          </label>
          <textarea
            className="form-control"
            id="descriptionTextArea"
            onChange={[Function]}
            rows="3"
            value={Array []}
          />
          <small
            className="form-text text-muted"
            id="descriptionHelp"
          >
            Enter your step-by-step Instructions.
          </small>
        </div>
        <div
          className="form-group col-4"
        >
          <button
            className="btn btn-outline-success btn-add"
          >
            Submit Recipe
          </button>
        </div>
      </div>
    </form>
  </main>
</div>
`;

exports[`Should set recipe category on input change 1`] = `
<div>
  <main
    className="container"
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          htmlFor="recipename"
        >
          Recipe Title
        </label>
        <input
          aria-describedby="recipeNameHelp"
          className="form-control"
          id="recipename"
          onChange={[Function]}
          type="text"
          value=""
        />
        <small
          className="form-text text-muted"
          id="recipeNameHelp"
        >
          Enter a short, catchy title for your Recipe.
        </small>
      </div>
      <div
        className="form-row"
      >
        <div
          className="form-group col-8"
        >
          <label
            htmlFor="inputDescription"
          >
            Description
          </label>
          <textarea
            className="form-control"
            id="descriptionTextArea"
            onChange={[Function]}
            rows="3"
            value=""
          />
          <small
            className="form-text text-muted"
            id="descriptionHelp"
          >
            Try to keep your description short and interesting.
          </small>
        </div>
        <div
          className="form-group col-4 upload-btn-wrapper"
        >
          <button
            className="btns"
          >
            Upload an Image
          </button>
          <input
            name="myfile"
            type="file"
          />
        </div>
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="recipename"
        >
          Category
        </label>
        <input
          aria-describedby="recipeNameHelp"
          className="form-control"
          id="recipename"
          onChange={[Function]}
          placeholder="Breakfast, Dinner, Dessert, Lunch"
          type="text"
          value="Lunch"
        />
        <small
          className="form-text text-muted"
          id="recipeNameHelp"
        >
          Enter a short, catchy title for your Recipe.
        </small>
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="recipename"
        >
          Ingredients
        </label>
        <input
          aria-describedby="recipeNameHelp"
          className="form-control"
          id="recipename"
          onChange={[Function]}
          type="text"
          value={Array []}
        />
        <small
          className="form-text text-muted"
          id="recipeNameHelp"
        >
          Enter your recipe ingredients, separated by Commas.
        </small>
      </div>
      <div
        className="form-row"
      >
        <div
          className="form-group col-8"
        >
          <label
            htmlFor="inputDescription"
          >
            Instructions
          </label>
          <textarea
            className="form-control"
            id="descriptionTextArea"
            onChange={[Function]}
            rows="3"
            value={Array []}
          />
          <small
            className="form-text text-muted"
            id="descriptionHelp"
          >
            Enter your step-by-step Instructions.
          </small>
        </div>
        <div
          className="form-group col-4"
        >
          <button
            className="btn btn-outline-success btn-add"
          >
            Submit Recipe
          </button>
        </div>
      </div>
    </form>
  </main>
</div>
`;

exports[`Should set recipe description on input change 1`] = `
<div>
  <main
    className="container"
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          htmlFor="recipename"
        >
          Recipe Title
        </label>
        <input
          aria-describedby="recipeNameHelp"
          className="form-control"
          id="recipename"
          onChange={[Function]}
          type="text"
          value=""
        />
        <small
          className="form-text text-muted"
          id="recipeNameHelp"
        >
          Enter a short, catchy title for your Recipe.
        </small>
      </div>
      <div
        className="form-row"
      >
        <div
          className="form-group col-8"
        >
          <label
            htmlFor="inputDescription"
          >
            Description
          </label>
          <textarea
            className="form-control"
            id="descriptionTextArea"
            onChange={[Function]}
            rows="3"
            value="Amazing Fried Chicken"
          />
          <small
            className="form-text text-muted"
            id="descriptionHelp"
          >
            Try to keep your description short and interesting.
          </small>
        </div>
        <div
          className="form-group col-4 upload-btn-wrapper"
        >
          <button
            className="btns"
          >
            Upload an Image
          </button>
          <input
            name="myfile"
            type="file"
          />
        </div>
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="recipename"
        >
          Category
        </label>
        <input
          aria-describedby="recipeNameHelp"
          className="form-control"
          id="recipename"
          onChange={[Function]}
          placeholder="Breakfast, Dinner, Dessert, Lunch"
          type="text"
          value=""
        />
        <small
          className="form-text text-muted"
          id="recipeNameHelp"
        >
          Enter a short, catchy title for your Recipe.
        </small>
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="recipename"
        >
          Ingredients
        </label>
        <input
          aria-describedby="recipeNameHelp"
          className="form-control"
          id="recipename"
          onChange={[Function]}
          type="text"
          value={Array []}
        />
        <small
          className="form-text text-muted"
          id="recipeNameHelp"
        >
          Enter your recipe ingredients, separated by Commas.
        </small>
      </div>
      <div
        className="form-row"
      >
        <div
          className="form-group col-8"
        >
          <label
            htmlFor="inputDescription"
          >
            Instructions
          </label>
          <textarea
            className="form-control"
            id="descriptionTextArea"
            onChange={[Function]}
            rows="3"
            value={Array []}
          />
          <small
            className="form-text text-muted"
            id="descriptionHelp"
          >
            Enter your step-by-step Instructions.
          </small>
        </div>
        <div
          className="form-group col-4"
        >
          <button
            className="btn btn-outline-success btn-add"
          >
            Submit Recipe
          </button>
        </div>
      </div>
    </form>
  </main>
</div>
`;

exports[`Should set the instructions on input change 1`] = `
<div>
  <main
    className="container"
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          htmlFor="recipename"
        >
          Recipe Title
        </label>
        <input
          aria-describedby="recipeNameHelp"
          className="form-control"
          id="recipename"
          onChange={[Function]}
          type="text"
          value=""
        />
        <small
          className="form-text text-muted"
          id="recipeNameHelp"
        >
          Enter a short, catchy title for your Recipe.
        </small>
      </div>
      <div
        className="form-row"
      >
        <div
          className="form-group col-8"
        >
          <label
            htmlFor="inputDescription"
          >
            Description
          </label>
          <textarea
            className="form-control"
            id="descriptionTextArea"
            onChange={[Function]}
            rows="3"
            value=""
          />
          <small
            className="form-text text-muted"
            id="descriptionHelp"
          >
            Try to keep your description short and interesting.
          </small>
        </div>
        <div
          className="form-group col-4 upload-btn-wrapper"
        >
          <button
            className="btns"
          >
            Upload an Image
          </button>
          <input
            name="myfile"
            type="file"
          />
        </div>
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="recipename"
        >
          Category
        </label>
        <input
          aria-describedby="recipeNameHelp"
          className="form-control"
          id="recipename"
          onChange={[Function]}
          placeholder="Breakfast, Dinner, Dessert, Lunch"
          type="text"
          value=""
        />
        <small
          className="form-text text-muted"
          id="recipeNameHelp"
        >
          Enter a short, catchy title for your Recipe.
        </small>
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="recipename"
        >
          Ingredients
        </label>
        <input
          aria-describedby="recipeNameHelp"
          className="form-control"
          id="recipename"
          onChange={[Function]}
          type="text"
          value={Array []}
        />
        <small
          className="form-text text-muted"
          id="recipeNameHelp"
        >
          Enter your recipe ingredients, separated by Commas.
        </small>
      </div>
      <div
        className="form-row"
      >
        <div
          className="form-group col-8"
        >
          <label
            htmlFor="inputDescription"
          >
            Instructions
          </label>
          <textarea
            className="form-control"
            id="descriptionTextArea"
            onChange={[Function]}
            rows="3"
            value={
              Array [
                "Parboil the rice for 20mins",
                "Boil the yam for 1hr.",
                "Do not set the kitchen on fire.",
              ]
            }
          />
          <small
            className="form-text text-muted"
            id="descriptionHelp"
          >
            Enter your step-by-step Instructions.
          </small>
        </div>
        <div
          className="form-group col-4"
        >
          <button
            className="btn btn-outline-success btn-add"
          >
            Submit Recipe
          </button>
        </div>
      </div>
    </form>
  </main>
</div>
`;

exports[`Should set the recipe ingredients on input change 1`] = `
<div>
  <main
    className="container"
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          htmlFor="recipename"
        >
          Recipe Title
        </label>
        <input
          aria-describedby="recipeNameHelp"
          className="form-control"
          id="recipename"
          onChange={[Function]}
          type="text"
          value=""
        />
        <small
          className="form-text text-muted"
          id="recipeNameHelp"
        >
          Enter a short, catchy title for your Recipe.
        </small>
      </div>
      <div
        className="form-row"
      >
        <div
          className="form-group col-8"
        >
          <label
            htmlFor="inputDescription"
          >
            Description
          </label>
          <textarea
            className="form-control"
            id="descriptionTextArea"
            onChange={[Function]}
            rows="3"
            value=""
          />
          <small
            className="form-text text-muted"
            id="descriptionHelp"
          >
            Try to keep your description short and interesting.
          </small>
        </div>
        <div
          className="form-group col-4 upload-btn-wrapper"
        >
          <button
            className="btns"
          >
            Upload an Image
          </button>
          <input
            name="myfile"
            type="file"
          />
        </div>
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="recipename"
        >
          Category
        </label>
        <input
          aria-describedby="recipeNameHelp"
          className="form-control"
          id="recipename"
          onChange={[Function]}
          placeholder="Breakfast, Dinner, Dessert, Lunch"
          type="text"
          value=""
        />
        <small
          className="form-text text-muted"
          id="recipeNameHelp"
        >
          Enter a short, catchy title for your Recipe.
        </small>
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="recipename"
        >
          Ingredients
        </label>
        <input
          aria-describedby="recipeNameHelp"
          className="form-control"
          id="recipename"
          onChange={[Function]}
          type="text"
          value={
            Array [
              "Curry",
              "Thyme",
              "Maggi",
              "Salt",
            ]
          }
        />
        <small
          className="form-text text-muted"
          id="recipeNameHelp"
        >
          Enter your recipe ingredients, separated by Commas.
        </small>
      </div>
      <div
        className="form-row"
      >
        <div
          className="form-group col-8"
        >
          <label
            htmlFor="inputDescription"
          >
            Instructions
          </label>
          <textarea
            className="form-control"
            id="descriptionTextArea"
            onChange={[Function]}
            rows="3"
            value={Array []}
          />
          <small
            className="form-text text-muted"
            id="descriptionHelp"
          >
            Enter your step-by-step Instructions.
          </small>
        </div>
        <div
          className="form-group col-4"
        >
          <button
            className="btn btn-outline-success btn-add"
          >
            Submit Recipe
          </button>
        </div>
      </div>
    </form>
  </main>
</div>
`;
